app/PostComponent.tsx

```diff
@@ -0,0 +1,105 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { FiHeart as HeartIcon } from 'react-icons/fi';
+import { GoPaperAirplane as ShareIcon } from 'react-icons/go';
+import { LuCircle as ProfileIcon } from 'react-icons/lu';
+import { supabase } from '@/supabase/client';
+import styles from './styles.module.css';
+import '../assets/global.css';
+
+interface Post {
+  author: string;
+  date: string;
+  image_url: string;
+  content: string;
+  likes: number;
+}
+
+export default function PostComponent() {
+  // Define a state variable to store the fetched data
+  const [postData, setPostData] = useState<Post[]>([]);
+
+  async function fetchPostData() {
+    // TODO: Fetch data using the supabase client
+    const response = await supabase.from('posts').select('*');
+    const { data, error } = response;
+    if (error) {
+      throw error;
+    }
+    const formattedData: Post[] = data.map(post => ({
```

Comment:
I think you might have done this to get around some typescript error? Instead you should be able to omit this map and directly do `setPostData(data as Post[])` to cast the type.